components/animation/index.js

```diff
@@ -131,35 +141,33 @@ AFRAME.registerComponent('animation', {
     data.pauseEvents.map(function (eventName) {
       el.removeEventListener(eventName, self.pauseAnimationBound);
     });
+    data.resumeEvents.map(function (eventName) {
+      el.removeEventListener(eventName, self.resumeAnimationBound);
+    });
+    data.restartEvents.map(function (eventName) {
+      el.removeEventListener(eventName, self.restartAnimationBound);
+    });
   },
 
   playAnimation: function () {
```

Comment:
Since the play/resume/restart animation methods are only one line long, we get rid of them and just call them directly